types/d3-collection/index.d.ts

```diff
@@ -2,11 +2,12 @@
 // Project: https://github.com/d3/d3-collection/
 // Definitions by: Tom Wanzek <https://github.com/tomwanzek>, Alex Ford <https://github.com/gustavderdrache>, Boris Yankov <https://github.com/borisyankov>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
```

Comment:
Out-of-memory bug fixed by setting this to 2.4.

Reply:
IMHO wouldn't it be better to fix whatever underlying bug causes the issue? There does not seem to be anything too exotic about the definitions. While I can confirm that bumping the TS version to a minimum of 2.4 circumvents the OoM bug, I'm wondering whether we are aiming at the wrong target.

Reply:
I don't think we would publish a new `typescript@2.3` release to fix a bug like this.

Reply:
Alright, I thought the reason for the OoM was buried somewhere in the dtslint workings. I noticed that other Definitions (e.g. d3-geo) have had TS 2.3 set for quite some time without any issue. So, if it's TS 2.3 itself, then yep we'll have to bite the bullet and publish as >=TS 2.4.